lib/std/net.zig

```diff
@@ -1591,16 +1591,14 @@ const ResolvConf = struct {
     ndots: u32,
     timeout: u32,
     search: ArrayList(u8),
-    /// TODO there are actually only allowed to be maximum 3 nameservers, no need
-    /// for an array list.
-    ns: ArrayList(LookupAddr),
+    ns: [3]?LookupAddr,
 
     /// Returns `error.StreamTooLong` if a line is longer than 512 bytes.
     /// TODO: https://github.com/ziglang/zig/issues/2765 and https://github.com/ziglang/zig/issues/2761
     fn init(rc: *ResolvConf, gpa: Allocator) !void {
```

Comment:
Since gpa is now only used in 2 functions, `deinit` and `parse`, it makes sense to pass it directly to those two and not store it in struct itself.
```suggestion
    fn init(rc: *ResolvConf) !void {
```